packages/app/cypress/e2e/integration/runs.spec.ts

```diff
@@ -1,80 +1,249 @@
-describe('App: Runs Page', () => {
+import type { Interception } from '@packages/net-stubbing/lib/external-types'
+
+describe('App: Runs', () => {
   beforeEach(() => {
     cy.scaffoldProject('component-tests')
     cy.openProject('component-tests')
     cy.startAppServer()
   })
 
-  it('resolves the runs page', () => {
-    cy.loginUser()
-    cy.visitApp()
-    cy.wait(1000)
-    cy.get('[href="#/runs"]').click()
-    cy.get('[data-cy="runs"]')
+  context('Runs Page', () => {
+    it('resolves the runs page', () => {
+      cy.loginUser()
+      cy.visitApp()
+      cy.wait(1000)
+      cy.get('[href="#/runs"]').click()
+      cy.get('[data-cy="runs"]')
+      cy.get('[data-cy="app-header-bar"]').findByText('Runs').should('be.visible')
+    })
+
+    it('shows the loader', () => {
+      cy.loginUser()
+      cy.remoteGraphQLIntercept(async (obj) => {
+        await new Promise((resolve) => setTimeout(resolve, 200))
+
+        return obj.result
+      })
+
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.get('[data-cy="runs-loader"]')
+      cy.get('[data-cy="runs"]')
+    })
   })
 
-  it('shows the loader', () => {
-    cy.loginUser()
-    cy.remoteGraphQLIntercept(async (obj) => {
-      await new Promise((resolve) => setTimeout(resolve, 200))
+  context('Runs - Login', () => {
+    it('when logged out, shows call to action', () => {
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.contains('Log In').should('exist')
+    })
 
-      return obj.result
+    it('clicking the login button will open the login modal', () => {
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.contains('Log In').click()
+      cy.get('[id="headlessui-dialog-title-13"]').contains('Log in to Cypress')
```

Comment:
This might get flaky depending on how this title is id is generated. I'd combine lines 46 and 47 into something like:

```js
cy.findByRole('dialog', { name: 'Log in to Cypress' }).within(() => {
  cy.get('button').contains('Log In')
})
```